src/game/PetAI.cpp

```diff
@@ -86,6 +86,18 @@ void PetAI::UpdateAI(const uint32 diff)
     else
         m_updateAlliesTimer -= diff;
 
+    // Prevent losing the pet action bar after reloading the user interface
```

Comment:
I breakpointed every unhandled opcode :P

And you can also fix the issue for mindcontrol as well by doing this:
if(_player->GetPet())
      _player->PetSpellInitialize();
else if(_player->GetCharm())
      _player->CharmSpellInitialize();